File: 1574-MaximumProductOfTwoElementsInAnArray/1574-MaximumProductOfTwoElementsInAnArray.java
// Last updated: 8/19/2026, 10:37:06 AM
class Solution {
    public int maxProduct(int[] nums) {
        int max1 = 1, max2 = 1;

        for(int ele : nums){
            if(max1 <= ele){
                max2 = max1;
                max1 = ele;
            }
            else if(max2 <= ele){
                max2 = ele;
            }
        }

        return (max1 - 1) * (max2 - 1);
    }
}

File: 1574-MaximumProductOfTwoElementsInAnArray/1574-MaximumProductOfTwoElementsInAnArray.py
# Last updated: 8/19/2026, 10:37:10 AM
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        maxi=0
        maxii=0
        for i in range(len(nums)):
            if nums[i]>maxi:
                maxii=maxi
                maxi=nums[i]
            elif nums[i]>maxii:
                maxii=nums[i]
        return (maxi-1)*(maxii-1)
        
    
